plugin/pkg/scheduler/algorithm/predicates/predicates_test.go

```diff
@@ -1207,13 +1210,14 @@ func TestServiceAffinity(t *testing.T) {
 	node4 := api.Node{ObjectMeta: api.ObjectMeta{Name: "machine4", Labels: labels4}}
 	node5 := api.Node{ObjectMeta: api.ObjectMeta{Name: "machine5", Labels: labels4}}
 	tests := []struct {
-		pod      *api.Pod
-		pods     []*api.Pod
-		services []*api.Service
-		node     *api.Node
-		labels   []string
-		fits     bool
-		test     string
+		pod            *api.Pod
+		pods           []*api.Pod
+		services       []*api.Service
+		node           *api.Node
+		labels         []string
+		fits           bool
+		test           string
+		skipPrecompute bool
```

Comment:
As per comment above, IF this is specified as false, we can test that the function has the same results w/ or w/o precomputation.  That can be used later so that we can verify the function is 100% robust and independent of metadata, if we want to...